virttest/qemu_vm.py

```diff
@@ -4764,10 +4764,11 @@ def is_json_data(data):
                             v = parse_json(v)
                         if k != key:
                             continue
-                        if v != value:
-                            matched = False
+                        if v == value:
+                            matched += 1
+                        else:
                             break
-                if matched:
+                if matched > 0 and matched == len(p_dict.keys()):
```

Comment:
Maybe there is a simpler way to handle this issue and keep the original meaning of `matched`.
use` for ... else` statement
```
                for key, value in six.iteritems(p_dict):
                    if is_json_data(value):
                        value = parse_json(value)
                    for (k, v) in traverse_nested_dict(block):
                        if is_json_data(v):
                            v = parse_json(v)
                        if k != key:
                            continue
                        if v != value:
                            matched = False
                            break
                        else:
                            break
                    else:
                        matched = False
```